src/parser/scaleMatrix.ts

```diff
@@ -1,6 +1,17 @@
-//@ts-nocheck
+import { TMat2D } from '../typedefs';
 
-export function scaleMatrix(matrix, args) {
+/**
+ * A scale matrix
+ * Takes form
+ * [x 0 0]
+ * [0 y 0]
+ * [0 0 1]
+ * For more info, see
+ * @link https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/transform#scale
+ */
+type TMatScale = [x: number] | [x: number, y: number];
+
+export function scaleMatrix(matrix: TMat2D, args: TMatScale | number[]) {
```

Comment:
So i think the code is very old and bad.
I would argue we fix it in a followup since @Lazauya made a work on types and regex and i think we can focus on that.
But yes, definetely this matrix loop is wrong, it has a mutable array that gets renewed evey loop with a new one, in a let. Is just bad code.